sjasm/directives.cpp

```diff
@@ -1801,21 +1801,21 @@ void _lua_showerror() {
 	int ln;
 
 	// part from Error(...)
-	char *err = STRDUP(lua_tostring(LUA, -1));
+	char *err = STRDUP(lua_tostring(LUA, -1)), *lnp = err;
 	if (err == NULL) {
 		Error("No enough memory!", NULL, FATAL);
 	}
-	err += 18;
-	char *pos = strstr(err, ":");
-	*(pos++) = 0;
-	ln = atoi(err) + LuaLine;
+	while (*lnp != ':' || !isdigit((unsigned char) *(lnp+1))) lnp++;
+	char *msgp = strchr(++lnp, ':');
+	*(msgp++) = '\0';
+	ln = atoi(lnp) + LuaLine;
```

Comment:
with extra function all this code in _show... will turn into simple `ln = extract...(lua_tostring(LUA, -1));`